### DISCUSSION TITLE
Constructor dependencies in Command

### DISCUSSION THREAD

**Discussion by mrhx** [nutgram/nutgram#781] [Q&A] [answered]
Hi,

I've tried recently to create a simple Nutgram command using `SergiX44\Nutgram\Handlers\Type\Command` and constructor dependencies, but I've found that it is not possible because Nutgram uses simple `new $command()` to create new instances of commands in `MessageListeners::registerCommand()` instead of using its own built-in dependency container.

Is it intended to be like this?

Thanks!

**Comment by Lukasss93** [comment, marked as answer]
This is something we know.
In the meantime, you can use DI in the "handle" method or you can use the `onCommand` method to register your commands
